indi-orion-ssg3/orion_ssg3_ccd.cpp

```diff
@@ -0,0 +1,580 @@
+
+    LOG_INFO("Exposure complete.");
+}
+
+#define TEMP_THRESHOLD 0.25
+/**
+ * Set the CCD temperature 
+ */
+int SSG3CCD::SetTemperature(double temperature)
+{
+    LOGF_INFO("Setting temperature to %.2f C.", temperature);
+
+    TemperatureRequest = temperature;
+
+    if (std::abs(temperature - TemperatureN[0].value) < TEMP_THRESHOLD)
+        return 1;
+
+    if (!activateCooler(true)) {
+        return -1;
+    }
+
+    return 0;
+}
+
+bool SSG3CCD::activateCooler(bool enable)
+{
+    int rc;
+
+    IUResetSwitch(&CoolerSP);
+    if (enable)
+    {
+        rc = orion_ssg3_set_temperature(&ssg3, TemperatureRequest);
+        if (rc < 0) {
+            LOG_ERROR("Failed to turn on cooling.");
+            CoolerSP.setState(IPS_ALERT);
+            IDSetSwitch(&CoolerSP, nullptr);
+            return false;
+        }
+
+        CoolerSP[COOLER_ON].setState(ISS_ON);
+        CoolerSP[COOLER_OFF].setState(ISS_OFF);
+        CoolerSP.setState(IPS_OK);
+    } else {
+        rc = orion_ssg3_cooling_off(&ssg3);
+        if (rc < 0) {
+            LOG_ERROR("Failed to turn off cooling.");
+            CoolerSP.setState(IPS_ALERT);
+            IDSetSwitch(&CoolerSP, nullptr);
+            return false;
+        }
+
+        CoolerSP[COOLER_ON].setState(ISS_OFF);
+        CoolerSP[COOLER_OFF].setState(ISS_ON);
+        CoolerSP.setState(IPS_IDLE);
+    }
+
+    IDSetSwitch(&CoolerSP, nullptr);
+    return true;
+}
+
+void SSG3CCD::updateTemperature(void)
+{
+    float temp;
+    int rc;
+
+    rc = orion_ssg3_get_temperature(&ssg3, &temp);
+    if (rc < 0) {
+        TemperatureNP.s = IPS_ALERT;
+    } else {
+        LOGF_DEBUG("Read temperature: %f\n", temp);
+        TemperatureN[0].value = temp;
+        TemperatureNP.s = IPS_OK;
+    }
+    IDSetNumber(&TemperatureNP, nullptr);
+
+    if (CoolerSP[COOLER_ON].getState() == ISS_ON) {
+        rc = orion_ssg3_get_cooling_power(&ssg3, &temp);
+        if (rc < 0) {
+            CoolerPowerNP.setState(IPS_ALERT);
+        } else {
+            LOGF_DEBUG("Read temperature: %f\n", temp);
+            CoolerPowerNP[0].setValue(temp);
+            CoolerPowerNP.setState(IPS_OK);
+        }
+        IDSetNumber(&CoolerPowerNP, nullptr);
+    } else {
+        CoolerPowerNP[0].setValue(0);
+        CoolerPowerNP.setState(IPS_OK);
+        IDSetNumber(&CoolerPowerNP, nullptr);
+    }
+}
+
+void SSG3CCD::stopNSGuide()
+{
+    NSTimer.stop();
+    GuideComplete(AXIS_DE);
+}
+
+void SSG3CCD::stopWEGuide()
+{
+    WETimer.stop();
+    GuideComplete(AXIS_RA);
+}
+
+IPState SSG3CCD::GuideNorth(uint32_t ms)
+{
+    int rc;
+    rc = orion_ssg3_st4(&ssg3, SSG3_GUIDE_NORTH, ms);
+    if (rc) {
+        return IPS_ALERT;
+    }
+    NSTimer.callOnTimeout(std::bind(&SSG3CCD::stopNSGuide, this));
+    NSTimer.start(ms);
+    return IPS_BUSY;
+}
+
+IPState SSG3CCD::GuideSouth(uint32_t ms)
+{
+    int rc;
+    rc = orion_ssg3_st4(&ssg3, SSG3_GUIDE_SOUTH, ms);
+    if (rc) {
+        return IPS_ALERT;
+    }
```

Comment:
since rc is not used, you can simplify this:

if (orion_ssg3_st4(&ssg3, SSG3_GUIDE_SOUTH, ms))
   return IPS_ALERT;